src/screens/Editor/Editor.js

```diff
@@ -21,19 +21,40 @@ function findLocations(subject) {
   })
 }
 
-function Editor ({ match }) {
+function Editor ({ match, testTime }) {
   const store = React.useContext(AppContext)
   const editorBox = React.useRef(null)
 
   React.useEffect(() => {
+    let accessTime = new Date()
+
     const setResources = async () => {
       await store.subjects.fetchSubject(match.params.subject)
       await store.getResources(match.params)
     }
     setResources()
     store.transcriptions.setActiveTranscription()
     undoManager.clear()
-  }, [match, store])
+
+    function handleTimeCheck() {
+      let recheckTime = testTime || new Date(accessTime).setHours(accessTime.getHours() + 3)
+      const shouldRecheck = new Date() > recheckTime
+      if (shouldRecheck) {
+        accessTime = new Date()
+        recheckTime = new Date(accessTime).setHours(accessTime.getHours() + 3)
+        store.transcriptions.checkIfLocked()
+      }
+    }
+
+    window.addEventListener('beforeunload', store.transcriptions.unlockTranscription)
+    window.addEventListener('visibilitychange', handleTimeCheck)
```

Comment:
Just for my own clarity, these two actions of setting the resources and active transcription and then handling the time check and event listeners are in the same effect because you want these both to happen on mount and unmount?

Reply:
I only want them to happen on mount. Anything wrapped below in the `return () => { ... }` block will execute on unmount, which is removing the event listeners and unlocking the transcription. However, in the case the window is closed explicitly (clicking the browser 'X' button) without unmounting the component (eg. navigating to another part of the page) then this `beforeUnload` listener would kick in.

Reply:
Thanks for the order of operations explanation!